src/main/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java

```diff
@@ -44,262 +47,290 @@
-			webviewContainer.setPadding(10, 10, 10, 10);
-			webviewContainer.setBackgroundColor(R.drawable.warning_background);
-		}
-
-		container = (XWalkView) findViewById(R.id.wbvMain);
-
-		configureUseragent();
-
-		enableLocationUpdates();
-		setUpUiClient(container);
-		enableRemoteChromeDebugging();
-		enableJavascript(container);
-		enableStorage(container);
-
-		enableUrlHandlers(container);
-
-		browseToRoot();
-
-		if(settings.allowsConfiguration()) {
-			toast(redactUrl(appUrl));
-		}
-	}
-
-	@Override public boolean onCreateOptionsMenu(Menu menu) {
-		if(settings.allowsConfiguration()) {
-			getMenuInflater().inflate(R.menu.unbranded_web_menu, menu);
-		} else {
-			getMenuInflater().inflate(R.menu.web_menu, menu);
-		}
-		return super.onCreateOptionsMenu(menu);
-	}
+        super.onCreate(savedInstanceState);
+
+        trace(this, "Starting XWalk webview...");
+
+        this.simprints = new SimprintsSupport(this);
+        this.photoGrabber = new PhotoGrabber(this);
+        this.mrdt = new MrdtSupport(this);
+        try {
+            this.smsSender = new SmsSender(this);
+        } catch (Exception ex) {
+            error(ex, "Failed to create SmsSender.");
+        }
+
+        this.settings = SettingsStore.in(this);
+        this.appUrl = settings.getAppUrl();
+
+        this.requestWindowFeature(Window.FEATURE_NO_TITLE);
+        setContentView(R.layout.main);
+
+        // Add an alarming red border if using configurable (i.e. dev)
+        // app with a medic production server.
+        if (settings.allowsConfiguration() &&
+                appUrl.contains("app.medicmobile.org")) {
+            View webviewContainer = findViewById(R.id.lytWebView);
+            webviewContainer.setPadding(10, 10, 10, 10);
+            webviewContainer.setBackgroundColor(R.drawable.warning_background);
+        }
+
+        container = (XWalkView) findViewById(R.id.wbvMain);
+
+        configureUseragent();
+
+        enableLocationUpdates();
+        setUpUiClient(container);
+        enableRemoteChromeDebugging();
+        enableJavascript(container);
+        enableStorage(container);
+
+        enableUrlHandlers(container);
+
+        browseToRoot();
+
+        if (settings.allowsConfiguration()) {
+            toast(redactUrl(appUrl));
+        }
+
+        /*
+        * Created SharedPreferences A Store Procedure in Android
+        * In onStop() We are Storing the Current Path of the User
+        * When we relaunch the application we are loading Stored URL in the container i.e XWalkView
+        * */
+        sharedPreferences = getSharedPreferences(MY_LAST_ACTIVITY, 0); // 0 - for private mode
+        String recentNavigation = sharedPreferences.getString(MY_LAST_ACTIVITY, "");
+        if (!(recentNavigation != null && recentNavigation.isEmpty())) {
+            container.loadUrl(recentNavigation);
+            Log.d("Gopinath", sharedPreferences.getString(MY_LAST_ACTIVITY, ""));
```

Comment:
It looks like you left this in by mistake. Either remove the line or change the text description.

Reply:
Removed Log used for my testing purpose